lib/subordinateio.c

```diff
@@ -308,24 +326,25 @@ static bool have_range(struct commonio_db *db,
 	return false;
 }
 
-static bool append_range(struct subordinate_range ***ranges, const struct subordinate_range *new, int n)
+static bool append_range(struct subid_range ***ranges, const struct subordinate_range *new, int n)
 {
-	struct subordinate_range *tmp;
+	struct subid_range *tmp;
 	if (!*ranges) {
-		*ranges = malloc(sizeof(struct subordinate_range *));
+		*ranges = malloc(sizeof(struct subid_range *));
 		if (!*ranges)
 			return false;
 	} else {
-		struct subordinate_range **new;
-		new = realloc(*ranges, (n + 1) * (sizeof(struct subordinate_range *)));
+		struct subid_range **new;
+		new = realloc(*ranges, (n + 1) * (sizeof(struct subid_range *)));
 		if (!new)
 			return false;
 		*ranges = new;
 	}
 	(*ranges)[n] = NULL;
-	tmp = subordinate_dup(new);
+	tmp = malloc(sizeof(*tmp));
 	if (!tmp)
 		return false;
+	memcpy(tmp, new, sizeof(*tmp));
```

Comment:
That doesn't matter much already, since next commit fixes this...

but `tmp` was `subid_range *` and `new` was `subordinate_range *` with `owner` as a first field, so this `memcpy` was corrupting content of `tmp`...